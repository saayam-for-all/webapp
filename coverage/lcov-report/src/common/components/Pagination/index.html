<!doctype html>
<html lang="en">
  <head>
    <title>Code coverage report for src/common/components/Pagination</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <link
      rel="shortcut icon"
      type="image/x-icon"
      href="../../../../favicon.png"
    />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type="text/css">
      .coverage-summary .sorter {
        background-image: url(../../../../sort-arrow-sprite.png);
      }
    </style>
  </head>

  <body>
    <div class="wrapper">
      <div class="pad1">
        <h1>
          <a href="../../../../index.html">All files</a>
          src/common/components/Pagination
        </h1>
        <div class="clearfix">
          <div class="fl pad1y space-right2">
            <span class="strong">56.66% </span>
            <span class="quiet">Statements</span>
            <span class="fraction">17/30</span>
          </div>

          <div class="fl pad1y space-right2">
            <span class="strong">41.66% </span>
            <span class="quiet">Branches</span>
            <span class="fraction">10/24</span>
          </div>

          <div class="fl pad1y space-right2">
            <span class="strong">22.22% </span>
            <span class="quiet">Functions</span>
            <span class="fraction">2/9</span>
          </div>

          <div class="fl pad1y space-right2">
            <span class="strong">57.14% </span>
            <span class="quiet">Lines</span>
            <span class="fraction">16/28</span>
          </div>
        </div>
        <p class="quiet">
          Press <em>n</em> or <em>j</em> to go to the next uncovered block,
          <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
          <div class="quiet">
            Filter:
            <input type="search" id="fileSearch" />
          </div>
        </template>
      </div>
      <div class="status-line medium"></div>
      <div class="pad1">
        <table class="coverage-summary">
          <thead>
            <tr>
              <th data-col="file" data-fmt="html" data-html="true" class="file">
                File
              </th>
              <th
                data-col="pic"
                data-type="number"
                data-fmt="html"
                data-html="true"
                class="pic"
              ></th>
              <th
                data-col="statements"
                data-type="number"
                data-fmt="pct"
                class="pct"
              >
                Statements
              </th>
              <th
                data-col="statements_raw"
                data-type="number"
                data-fmt="html"
                class="abs"
              ></th>
              <th
                data-col="branches"
                data-type="number"
                data-fmt="pct"
                class="pct"
              >
                Branches
              </th>
              <th
                data-col="branches_raw"
                data-type="number"
                data-fmt="html"
                class="abs"
              ></th>
              <th
                data-col="functions"
                data-type="number"
                data-fmt="pct"
                class="pct"
              >
                Functions
              </th>
              <th
                data-col="functions_raw"
                data-type="number"
                data-fmt="html"
                class="abs"
              ></th>
              <th
                data-col="lines"
                data-type="number"
                data-fmt="pct"
                class="pct"
              >
                Lines
              </th>
              <th
                data-col="lines_raw"
                data-type="number"
                data-fmt="html"
                class="abs"
              ></th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td class="file medium" data-value="Pagination.jsx">
                <a href="Pagination.jsx.html">Pagination.jsx</a>
              </td>
              <td data-value="56.66" class="pic medium">
                <div class="chart">
                  <div class="cover-fill" style="width: 56%"></div>
                  <div class="cover-empty" style="width: 44%"></div>
                </div>
              </td>
              <td data-value="56.66" class="pct medium">56.66%</td>
              <td data-value="30" class="abs medium">17/30</td>
              <td data-value="41.66" class="pct low">41.66%</td>
              <td data-value="24" class="abs low">10/24</td>
              <td data-value="22.22" class="pct low">22.22%</td>
              <td data-value="9" class="abs low">2/9</td>
              <td data-value="57.14" class="pct medium">57.14%</td>
              <td data-value="28" class="abs medium">16/28</td>
            </tr>
          </tbody>
        </table>
      </div>
      <div class="push"></div>
      <!-- for sticky footer -->
    </div>
    <!-- /wrapper -->
    <div class="footer quiet pad2 space-top1 center small">
      Code coverage generated by
      <a
        href="https://istanbul.js.org/"
        target="_blank"
        rel="noopener noreferrer"
        >istanbul</a
      >
      at 2025-10-14T17:34:04.000Z
    </div>
    <script src="../../../../prettify.js"></script>
    <script>
      window.onload = function () {
        prettyPrint();
      };
    </script>
    <script src="../../../../sorter.js"></script>
    <script src="../../../../block-navigation.js"></script>
  </body>
</html>
