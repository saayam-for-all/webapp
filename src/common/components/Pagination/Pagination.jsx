import React from "react";
import { RiArrowLeftSLine, RiArrowRightSLine } from "react-icons/ri";
const Pagination = ({
  currentPage,
  totalPages,
  onPageChange,
  rowsPerPage,
  totalRows,
  onRowsPerPageChange,
}) => {
  const renderPageNumbers = () => {
    let pages = [];
    const maxFirstPages = 5;
    const pageRange = 2;

    // show the first page
    pages.push(
      <button
        key={1}
        onClick={() => onPageChange(1)}
        className={`px-3 py-1 border rounded ${
          1 === currentPage
            ? "bg-blue-500 text-white hover:bg-blue-500"
            : "bg-gray-200 text-black hover:bg-white"
        }`}
      >
        {1}
      </button>,
    );

    // show ... if there are more pages
    if (currentPage >= maxFirstPages) {
      pages.push(
        <span key="ellipsis-start" className="px-3 py-1">
          ...
        </span>,
      );
      for (let i = currentPage - pageRange; i <= currentPage + pageRange; i++) {
        if (i > 1 && i < totalPages) {
          pages.push(
            <button
              key={i}
              onClick={() => onPageChange(i)}
              className={`px-3 py-1 border rounded ${
                i === currentPage
                  ? "bg-blue-500 text-white hover:bg-blue-500"
                  : "bg-gray-200 text-black hover:bg-white"
              }`}
            >
              {i}
            </button>,
          );
        }
      }
    } else {
      if (totalPages > 2) {
        console.log("more");
        for (let i = 2; i <= Math.min(maxFirstPages, totalPages - 1); i++) {
          pages.push(
            <button
              key={i}
              onClick={() => onPageChange(i)}
              className={`px-3 py-1 border rounded ${
                i === currentPage
                  ? "bg-blue-500 text-white hover:bg-blue-500"
                  : "bg-gray-200 text-black hover:bg-white"
              }`}
            >
              {i}
            </button>,
          );
        }
      }
    }

    if (currentPage + pageRange < totalPages) {
      pages.push(
        <span key="ellipsis-end" className="px-3 py-1">
          ...
        </span>,
      );
    }

    if (pages.length < totalPages) {
      // show the last page
      pages.push(
        <button
          key={totalPages}
          onClick={() => onPageChange(totalPages)}
          className={`px-3 py-1 border rounded ${
            totalPages === currentPage
              ? "bg-blue-500 text-white hover:bg-blue-500"
              : "bg-gray-200 text-black hover:bg-white"
          }`}
        >
          {totalPages}
        </button>,
      );
    }
    return pages;
  };

  return (
    <div className="p-5">
      <div className="flex justify-between items-center">
        <div className="flex items-center justify-center mb-2">
          <span className="mr-2 text-gray-400">
            Showing data {(currentPage - 1) * rowsPerPage + 1}-
            {Math.min(currentPage * rowsPerPage, totalRows)} of {totalRows}{" "}
            entries
          </span>
          <div className="flex items-center ml-2">
            <label htmlFor="rowsPerPage" className="mr-2">
              Rows per view:
            </label>
            <select
              id="rowsPerPage"
              value={rowsPerPage}
              onChange={(e) => onRowsPerPageChange(Number(e.target.value))}
              className="border rounded px-2 py-1"
            >
              <option value={5}>5 rows</option>
              <option value={10}>10 rows</option>
              <option value={20}>20 rows</option>
            </select>
          </div>
        </div>
        <div className="flex items-center space-x-2">
          <button
            onClick={() => onPageChange(Math.max(currentPage - 1, 1))}
            disabled={currentPage === 1}
            className={`bg-gray-200 text-black py-2 px-2 rounded hover:bg-gray-500 ${currentPage === 1 && "invisible"}`}
          >
            <RiArrowLeftSLine />
          </button>
          {renderPageNumbers()}
          <button
            onClick={() => onPageChange(Math.min(currentPage + 1, totalPages))}
            disabled={currentPage === totalPages}
            className={`bg-gray-200 text-black py-2 px-2 rounded hover:bg-gray-500 ${currentPage === totalPages && "invisible"}`}
          >
            <RiArrowRightSLine />
          </button>
        </div>
      </div>
    </div>
  );
};

export default Pagination;
