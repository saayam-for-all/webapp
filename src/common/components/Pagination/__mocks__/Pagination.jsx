export default function Pagination({
  currentPage,
  totalPages,
  onPageChange,
  rowsPerPage,
  totalRows,
}) {
  return (
    <mock-pagination
      currentPage={currentPage}
      totalPages={totalPages}
      rowsPerPage={rowsPerPage}
      totalRows={totalRows}
    />
  );
}
