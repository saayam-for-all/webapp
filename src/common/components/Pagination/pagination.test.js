import { render } from "@testing-library/react";
import Pagination from "./Pagination";

describe("Pagination", () => {
  it("renders correctly", () => {
    const tree = render(
      <Pagination
        currentPage={1}
        totalPages={3}
        onPageChange={() => {}}
        rowsPerPage={10}
        totalRows={30}
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
