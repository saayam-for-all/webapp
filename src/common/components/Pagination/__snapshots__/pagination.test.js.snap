// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-5"
      >
        <div
          class="flex justify-between items-center"
        >
          <div
            class="flex items-center justify-center mb-2"
          >
            <span
              class="mr-2 text-gray-400"
            >
              Showing data 
              1
              -
              10
               of 
              30
               
              entries
            </span>
            <div
              class="flex items-center ml-2"
            >
              <label
                class="mr-2"
                for="rowsPerPage"
              >
                Rows per view:
              </label>
              <select
                class="border rounded px-2 py-1"
                id="rowsPerPage"
              >
                <option
                  value="5"
                >
                  5 rows
                </option>
                <option
                  value="10"
                >
                  10 rows
                </option>
                <option
                  value="20"
                >
                  20 rows
                </option>
              </select>
            </div>
          </div>
          <div
            class="flex items-center space-x-2"
          >
            <button
              class="bg-gray-200 text-black py-2 px-2 rounded hover:bg-gray-500 invisible"
              disabled=""
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10.8284 12.0007L15.7782 16.9504L14.364 18.3646L8 12.0007L14.364 5.63672L15.7782 7.05093L10.8284 12.0007Z"
                />
              </svg>
            </button>
            <button
              class="px-3 py-1 border rounded bg-blue-500 text-white hover:bg-blue-500"
            >
              1
            </button>
            <button
              class="px-3 py-1 border rounded bg-gray-200 text-black hover:bg-white"
            >
              2
            </button>
            <button
              class="px-3 py-1 border rounded bg-gray-200 text-black hover:bg-white"
            >
              3
            </button>
            <button
              class="bg-gray-200 text-black py-2 px-2 rounded hover:bg-gray-500 false"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.1717 12.0007L8.22192 7.05093L9.63614 5.63672L16.0001 12.0007L9.63614 18.3646L8.22192 16.9504L13.1717 12.0007Z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-5"
    >
      <div
        class="flex justify-between items-center"
      >
        <div
          class="flex items-center justify-center mb-2"
        >
          <span
            class="mr-2 text-gray-400"
          >
            Showing data 
            1
            -
            10
             of 
            30
             
            entries
          </span>
          <div
            class="flex items-center ml-2"
          >
            <label
              class="mr-2"
              for="rowsPerPage"
            >
              Rows per view:
            </label>
            <select
              class="border rounded px-2 py-1"
              id="rowsPerPage"
            >
              <option
                value="5"
              >
                5 rows
              </option>
              <option
                value="10"
              >
                10 rows
              </option>
              <option
                value="20"
              >
                20 rows
              </option>
            </select>
          </div>
        </div>
        <div
          class="flex items-center space-x-2"
        >
          <button
            class="bg-gray-200 text-black py-2 px-2 rounded hover:bg-gray-500 invisible"
            disabled=""
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.8284 12.0007L15.7782 16.9504L14.364 18.3646L8 12.0007L14.364 5.63672L15.7782 7.05093L10.8284 12.0007Z"
              />
            </svg>
          </button>
          <button
            class="px-3 py-1 border rounded bg-blue-500 text-white hover:bg-blue-500"
          >
            1
          </button>
          <button
            class="px-3 py-1 border rounded bg-gray-200 text-black hover:bg-white"
          >
            2
          </button>
          <button
            class="px-3 py-1 border rounded bg-gray-200 text-black hover:bg-white"
          >
            3
          </button>
          <button
            class="bg-gray-200 text-black py-2 px-2 rounded hover:bg-gray-500 false"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.1717 12.0007L8.22192 7.05093L9.63614 5.63672L16.0001 12.0007L9.63614 18.3646L8.22192 16.9504L13.1717 12.0007Z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
